src/modules/powerrename/lib/Settings.cpp

```diff
@@ -470,12 +359,113 @@ void CRenameMRU::_FreeMRUList()
     }
 }
 
+CSettings::CSettings()
+{
+    std::wstring result = PTSettingsHelper::get_module_save_folder_location(L"PowerRename");
+    jsonFilePath = result + L"\\" + std::wstring(c_powerRenameDataFilePath);
+}
+
+void CSettings::LoadPowerRenameData()
+{
+    if (!std::filesystem::exists(jsonFilePath))
+    {
+        MigrateSettingsFromRegistry();
+
+        SavePowerRenameData();
+    }
+    else
+    {
+        ParseJsonSettings();
+    }
+}
+
+void CSettings::SavePowerRenameData() const
+{
+    json::JsonObject jsonData;
+
+    jsonData.SetNamedValue(c_enabled,                 json::value(settings.enabled));
+    jsonData.SetNamedValue(c_showIconOnMenu,          json::value(settings.showIconOnMenu));
+    jsonData.SetNamedValue(c_extendedContextMenuOnly, json::value(settings.extendedContextMenuOnly));
+    jsonData.SetNamedValue(c_persistState,            json::value(settings.persistState));
+    jsonData.SetNamedValue(c_mruEnabled,              json::value(settings.MRUEnabled));
+    jsonData.SetNamedValue(c_maxMRUSize,              json::value(settings.maxMRUSize));
+    jsonData.SetNamedValue(c_flags,                   json::value(settings.flags));
+    jsonData.SetNamedValue(c_searchText,              json::value(settings.searchText));
+    jsonData.SetNamedValue(c_replaceText,             json::value(settings.replaceText));
+
+    json::to_file(jsonFilePath, jsonData);

```

Comment:
`json::to_file` can throw exceptions, right?

Reply:
No, `json::from_file` can, and it has try-catch block within it.

Reply:
`ofstream` doesn't throw an exception if you pass an invalid file name or if it can't access the file?

Reply:
No, the do set error code, but no exception:
```
Constructs an ofstream object, initially associated with the file identified by its first argument (filename), open with the mode specified by mode.
Internally, its ostream base constructor is passed a pointer to a newly constructed filebuf object (the internal file stream buffer). Then, filebuf::open is called with filename and mode as arguments.
If the file cannot be opened, the stream's failbit flag is set.
```